website/client/js/services/userServices.js

```diff
@@ -103,8 +104,28 @@ angular.module('habitrpg')
 
           return Tasks.getUserTasks();
         })
+        // refresh all but completed todos
         .then(function (response) {
-          var tasks = response.data.data;
+          tasks = response.data.data;
+
+          // only refresh completed todos if the user has the completed tabs list open
+          var taskButton = document.getElementById('completedTasksButton');
```

Comment:
If possible I'd like to avoid relying on the DOM. Can you see if you can get access to the `list.view` value used here https://github.com/HabitRPG/habitrpg/pull/7673/files#diff-52e526ae61c107ffe0964693be9534faL34? Otherwise it's okay this way